Libs/DICOM/Core/ctkDICOMDatabase.cpp

```diff
@@ -206,7 +206,7 @@ QStringList ctkDICOMDatabasePrivate::allFilesInDatabase()
 
   while (allFilesQuery.next())
   {
-    allFileNames << this->absolutePathFromInternal(allFilesQuery.value(0).toString());
+    allFileNames << this->absolutePathFromInternalIfFile(allFilesQuery.value(0).toString());
```

Comment:
In general I agree with this, but since the term `fileName` is used in the schema I felt like keeping this consistency would lead to the most readable code.  For the reasons you list in your next comment I'm not sure `allUrls` is correct either.  Perhaps something generic like `instanceLocators` is more correct.  We could make a more sweeping change, including the schema, if we want to generalize.

Part of the reason for this is that some protocols don't use actual URLs to identify the concept of an "DICOM instance".  For example, DIMSE doesn't have URLs, but we might want to refer to an instance with something like `cget://<pacs ip>/<instance uid>`.  Currently I have to do this for AHI since frame-level access is a REST API call with dataset and image frame IDs as parameters, so I have a [synthetic "URL" string](https://github.com/pieper/DICOMLogic/blob/main/src/DICOMLogic/stores/DICOMAHIStore.py#L69-L74).